rs/boundary_node/rate_limits/canister/certification.rs

```diff
@@ -0,0 +1,174 @@
+use std::{cell::RefCell, rc::Rc};
+
+use ic_asset_certification::{Asset, AssetConfig, AssetEncoding, AssetFallbackConfig, AssetRouter};
+use ic_cdk::api::{data_certificate, set_certified_data};
+use ic_http_certification::{HeaderField, HttpCertificationTree, HttpRequest, HttpResponse};
+
+use crate::{
+    metrics::{export_metrics_as_http_response, with_metrics_registry},
+    state::with_canister_state,
+};
+
+thread_local! {
+    static HTTP_TREE: Rc<RefCell<HttpCertificationTree>> = Default::default();
+
+    // initializing the asset router with an HTTP certification tree is optional.
+    // if direct access to the HTTP certification tree is not needed for certifying
+    // requests and responses outside of the asset router, then this step can be skipped.
+    static ASSET_ROUTER: RefCell<AssetRouter<'static>> = RefCell::new(AssetRouter::with_tree(HTTP_TREE.with(|tree| tree.clone())));
+}
+
+// Handlers
+pub fn serve_asset(request: &HttpRequest<'static>) -> HttpResponse<'static> {
+    let path = match request.get_path() {
+        Ok(path) => path,
+        Err(err) => {
+            return HttpResponse::builder()
+                .with_status_code(400)
+                .with_body(
+                    format!("Malformed request: {:?}", err.to_string())
+                        .as_bytes()
+                        .to_vec(),
+                )
+                .build()
+        }
+    };
+
+    match path.as_str() {
+        // Serve non-certified metrics
+        "/metrics" => with_canister_state(|state| {
```

Comment:
Note to self, add the ability to add paths for skipping certification to the library, then this can be served on non-raw endpoints.